### PR TITLE
feat: introduce support for native astro integration package

### PR DESCRIPTION
**Pull request by lirantal**
Implement #345 - astro integration

Signed-off-by: Liran Tal <liran.tal@gmail.com>

### CODE REVIEW

File: packages/astro/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License Copyright (c) 2023 Liran Tal
```
Comment (ericallam) [inline packages/astro/LICENSE]: This made me realize that we don't have individual licenses currently for our public packages, which has been fixed in this commit:

https://github.com/triggerdotdev/trigger.dev/commit/b5db9f5e5cf121e37b6350bf28fb64dc81ec949b

Do you think you could update this LICENSE file to match the others with our copyright? I think our policy will be that if it's in the monorepo then the LICENSE should be under Trigger.dev. If that doesn't work for you, then it's also okay for it to live as a standalone package (although not under the `@trigger.dev` npm org.

File: packages/astro/main.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+export function createAstroRoute(client) {
```
Comment (ericallam) [inline packages/astro/main.js:1]: Ideally this package would be written in TypeScript to most of our other code

File: packages/astro/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Trigger.dev native integration add-on for Astro
+
+This Astro integration project provides a `createAstroRoute` function that can be used to create an API endpoint route in your Astro project for integration with the hosted Trigger.dev platform.
+
+## Usage
+
+1. Install the package:
+
+```bash
+npm install --save https://github.com/lirantal/triggerdev-astro-integration
```
Comment (ericallam) [inline packages/astro/README.md]: These instructions should be updated to reference the final package `@trigger.dev/astro`

File: packages/astro/package.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  "engines": {
+    "node": ">=18.0.0"
+  },
+  "license": "MIT",
+  "publishConfig": {
+    "access": "public"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/lirantal/triggerdev-astro-integration.git"
```
Comment (ericallam) [inline packages/astro/package.json]: These urls should be updated as well

**Review by ericallam — CHANGES_REQUESTED** [review]
Thanks for this, it's a great start! Just added a few comments, let me know what you think.

File: packages/astro/README.md (hunk 3 above)
Comment (lirantal) [reply packages/astro/README.md]: Fixed

File: packages/astro/LICENSE (hunk 1 above)
Comment (lirantal) [reply packages/astro/LICENSE]: Yes, fixed!

File: packages/astro/main.js (hunk 2 above)
Comment (lirantal) [reply packages/astro/main.js:1]: @ericallam I'm going to leave that part for you folks as TS isn't my thing and you know best how and what you want to do  ;-)

File: packages/astro/package.json (hunk 4 above)
Comment (lirantal) [reply packages/astro/package.json]: Fixed

**Comment by lirantal** [comment]
@ericallam addressed your comments, excluding the TS part.

**Review by ericallam — APPROVED** [review]

**Pull request merged**